File: DSA/Khushi_590028673/Day33/Q2.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

class Solution {
public:
    string postfixToPrefix(string s) {
        stack<string> st;

        for (char ch : s) {
            // If operand
            if ((ch >= 'A' && ch <= 'Z') || (ch >= 'a' && ch <= 'z') || (ch >= '0' && ch <= '9')) {
                st.push(string(1, ch));
            }
            // If operator
            else {
                string op2 = st.top();
                st.pop();

                string op1 = st.top();
                st.pop();

                string temp = ch + op1 + op2;
                st.push(temp);
            }
        }

        return st.top();
    }
};

int main() {
    Solution obj;

    string s;
    cout << "Enter postfix expression: ";
    cin >> s;

    cout << "Prefix Expression: " << obj.postfixToPrefix(s) << endl;

    return 0;
}

File: DSA/Navneet_590012555/Day33/Q2.java
String postfixToPrefix(String s) {
    Stack<String> st = new Stack<>();

    for (char c : s.toCharArray()) {
        if (Character.isLetter(c)) {
            st.push(c + "");
        } else {
            String a = st.pop();
            String b = st.pop();
            st.push(c + b + a);
        }
    }

    return st.peek();
}